lib/master.go

```diff
@@ -435,25 +452,21 @@ func (m *DefaultBaseMaster) OnError(err error) {
 	}
 }
 
+// refreshMetadata load and update metadata by current nodeID, advertiseAddr, etc.
 // master meta is persisted before it is created, in this function we update some
-// fileds to the current value, including epoch, nodeID and advertiseAddr.
+// fileds to the current value, including nodeID and advertiseAddr.
 func (m *DefaultBaseMaster) refreshMetadata(ctx context.Context) (isInit bool, epoch Epoch, err error) {
 	metaClient := NewMasterMetadataClient(m.id, m.metaKVClient)
 
 	masterMeta, err := metaClient.Load(ctx)
 	if err != nil {
 		return false, 0, err
 	}
-
-	epoch, err = metaClient.GenerateEpoch(ctx)
-	if err != nil {
-		return false, 0, errors.Trace(err)
-	}
+	epoch = masterMeta.Epoch
```

Comment:
Previous epoch logic depends on the increasing logic clock(epoch) of metastore(etcd).
Since we migrate to metaclient.KVClient from metadata.MetaKV, so I move the epoch refresh mechanism to jobmanager which can use the increasing logic clock(epoch) of embedded etcd.

Reply:
It breaks the semantic of epoch, since the master meta is updated here, but epoch remains unchanged.

Reply:
I'm not sure whether any side effect would be introduced.

Reply:
Already add some notice